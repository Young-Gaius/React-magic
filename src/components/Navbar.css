/* Navbar.css */
.navbar {
  background-color: #1f2937;
}

.navbar-content {
  margin: 0 auto;
  padding: 0 16px;
}

.navbar-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 64px;
  margin-bottom: 20px;
}

.navbar-logo {
  color: white;
  font-size: 24px;
  font-weight: bold;
  margin-right: 16px;
}

.navbar-links {
  display: flex;
  align-items: center;
}

.navbar-link {
  color: white;
  margin-left: 16px;
  transition: color 0.3s ease;
}

.navbar-link:hover {
  color: #d1d5db;
}
