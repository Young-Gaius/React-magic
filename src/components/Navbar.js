import React from 'react';
import { Link } from 'react-router-dom';
import './Navbar.css'; // Import the CSS file

function Navbar() {
  return (
    <nav className="navbar">
      <div className="navbar-content">
        <div className="navbar-container">
          <div className="navbar-logo">
            <Link to="/">Math Magicians</Link>
          </div>
          <div className="navbar-links">
            <Link to="/" className="navbar-link">
              Home
            </Link>
            <Link to="/calculator" className="navbar-link">
              Calculator
            </Link>
            <Link to="/quote" className="navbar-link">
              Quote
            </Link>
          </div>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
