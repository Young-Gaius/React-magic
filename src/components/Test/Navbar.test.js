import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

test('Navbar renders correctly', () => {
  const component = renderer.create(
    <Router>
      <Navbar />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
