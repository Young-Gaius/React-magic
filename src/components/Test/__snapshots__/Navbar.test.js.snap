// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-content"
  >
    <div
      className="navbar-container"
    >
      <div
        className="navbar-logo"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Math Magicians
        </a>
      </div>
      <div
        className="navbar-links"
      >
        <a
          className="navbar-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="navbar-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          className="navbar-link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </div>
</nav>
`;
